protocol/messenger_communities.go

```diff
@@ -2446,6 +2447,20 @@ func (m *Messenger) handleSyncCommunity(messageState *ReceivedMessageState, sync
 		}
 	}
 
+	id := crypto.CompressPubkey(orgPubKey)
+	savedCommunity, err := m.communitiesManager.GetByID(id)
+	if err != nil {
+		return err
+	}
+
+	if savedCommunity.HasPermissionToSendCommunityEvents() || savedCommunity.IsControlNode() {
+		err := m.handleCommunityTokensMetadata(savedCommunity.IDString(), savedCommunity.CommunityTokensMetadata())
```

Comment:
This function might be slow because it calls the chain sometimes, but `handleSyncCommunity` is already handled in the background, so I guess it's not an issue?

I'm just asking to be sure all is good in terms of responsiveness and timeliness.

Reply:
> This function might be slow because it calls the chain sometimes, but handleSyncCommunity is already handled in the background, so I guess it's not an issue?

Not sure if there's so much we can do about that. We will have to reach out to chain regardless, but in reality I think it's not going to be that often anyways.

Also, this is part of recovering an account which already has a 5(?)min timer. So it will be part of that. Sync at runtime indeed happens in the background.